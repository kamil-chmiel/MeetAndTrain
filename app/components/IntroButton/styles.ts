import styled, { css } from 'styled-components/native';

export const Container = styled.TouchableOpacity`
	height: 60;
	left: 0;
	right: 0;
	top: 50;
	align-items: center;
	border-color: ${(props) => props.color};
	border-width: 3;
	border-radius: 8;
	margin-bottom: 5;
	margin-top: 5;
`;

export const Title = styled.Text`
	flex: 1;
	margin-top: 14;
	font-size: 20;
	color: ${(props) => props.color};
	font-weight: bold;
`;
