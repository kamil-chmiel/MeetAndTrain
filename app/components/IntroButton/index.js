import React from 'react';
import { Image } from 'react-native';
import { Container, Title } from './styles';

const IntroButton = ({ title, color, onPress }) => {
	return (
		<Container color={color} onPress={onPress}>
			<Title color={color}>{title}</Title>
		</Container>
	);
};

export default IntroButton;
