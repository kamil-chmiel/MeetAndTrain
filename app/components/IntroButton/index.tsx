import React from 'react';
import { Container, Title } from './styles';

const IntroButton = ({ title, color = 'green', onPress }) => {
	return (
		<Container color={color} onPress={onPress}>
			<Title color={color}>{title}</Title>
		</Container>
	);
};

export default IntroButton;
