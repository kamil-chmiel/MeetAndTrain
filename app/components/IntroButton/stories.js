import React from 'react';
import { storiesOf } from '@storybook/react-native';

import IntroButton from './';

storiesOf('Components/IntroButton', module).add('initial', () => <IntroButton />);
