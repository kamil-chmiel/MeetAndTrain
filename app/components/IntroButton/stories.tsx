import React from 'react';
import { View } from 'react-native'
import { storiesOf } from '@storybook/react-native';

import IntroButton from '.';

storiesOf('Components/IntroButton', module)
.addDecorator(story => <View style={{ flex: 1, justifyContent: 'center' }}>{story()}</View>)
.add('initial', () => <IntroButton />)
.add('with text', () => <IntroButton title='test'/>)
.add('with custom color', () => <IntroButton color='red'/>)
