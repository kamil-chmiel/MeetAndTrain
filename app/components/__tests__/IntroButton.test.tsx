import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

import IntroButton from '../IntroButton'

const defaultProps = {
    color: "green",
}

it('should render correctly', () => {
    const tree = renderer.create(<IntroButton {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('should render correctly with white theme', () => {
    const tree = renderer.create(<IntroButton color="white" />).toJSON()
    expect(tree).toMatchSnapshot()
})