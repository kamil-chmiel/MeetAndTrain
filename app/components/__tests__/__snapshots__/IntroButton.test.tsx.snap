// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "green",
      "borderRadius": 8,
      "borderWidth": 3,
      "height": 60,
      "left": 0,
      "marginBottom": 5,
      "marginTop": 5,
      "opacity": 1,
      "right": 0,
      "top": 50,
    }
  }
>
  <Text
    color="green"
    style={
      Array [
        Object {
          "color": "green",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 14,
        },
      ]
    }
  />
</View>
`;

exports[`should render correctly with white theme 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "white",
      "borderRadius": 8,
      "borderWidth": 3,
      "height": 60,
      "left": 0,
      "marginBottom": 5,
      "marginTop": 5,
      "opacity": 1,
      "right": 0,
      "top": 50,
    }
  }
>
  <Text
    color="white"
    style={
      Array [
        Object {
          "color": "white",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 14,
        },
      ]
    }
  />
</View>
`;
